oam/install_scripts/mcs-savebrm.py.in

```
@@ -184,42 +197,191 @@ def is_node_primary(conf_root):
+    
+    return 'backup_{}'.format(epoch_time)
+
+
+def get_save_brm_dir_path(a_mcs_config_root):
+    """Returns a path that SM treats as local
 
-    if is_node_primary(config_root):
+    :param file_path: xml config XML root
+    :rtype
+    """
+    save_brm_dir_path = BRM_BACKUP_PATH
+    if a_mcs_config_root is not None:
         try:
-            retcode = subprocess.check_call(SAVEBRM, shell=True)
-        except subprocess.CalledProcessError as exc:
-            logging.error('{} exits with {}.'.format(exc.cmd, exc.returncode))
-            sys.exit(1)
+            system_temp_file_dir = a_mcs_config_root.find('./SystemConfig/SystemTempFileDir').text
+            hdfs_rdwr_scratch = a_mcs_config_root.find('./SystemConfig/hdfsRdwrScratch').text
+            # There is a danger to have no '/' in the end of system_temp_file_dir
+            # or have two of them there. In both cases save_brm will fail to store
+            # files locally.
+            save_brm_dir_path = system_temp_file_dir + hdfs_rdwr_scratch
+        except AttributeError:
+            logging.error('Exception while getting SystemTempFileDir and hdfsRdwrScratch from Columnstore.xml', exc_info=True)
+    
+    return save_brm_dir_path
+
+
+def get_save_brm_path_prefix(a_mcs_config_root):
+    """Returns a path that SM treats as local
+
+    :param file_path: xml config XML root
+    :rtype: String
+    """
+    epoch_prefix = get_epoch_prefix()
+    return get_save_brm_dir_path(a_mcs_config_root) + '/' + BRM_BACKUP_PATH_PART.format(epoch_prefix)
+
+
+def call_save_brm(path):
+    """Calls save_brm first and then tries to call it with local path. 
+
+    :param file_path: xml config XML root
+    :rtype: None
+    """
+    savebrm_cmd = SAVEBRM + ' ' + path
+    try:
+        subprocess.check_call(savebrm_cmd, shell=True)
+    except subprocess.CalledProcessError as exc:
+        logging.error('The call to {} exits with {}.'.format(savebrm_cmd, exc.returncode))
+        return None
```

Better to throw an Exception. It's more readable and clear.